### PR TITLE
[2.x.x] Rebuild 2.15.1 for for python312

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@maxbachmann, obviously, we'll see what CI has to say but maybe take a look if you have a chance to make sure this is what you suggested.

**Comment by maxbachmann** [comment]
Yes this was what I did mean. Apparently it is failing because of some of the breaking changes in later alpha releases.  I will look into it.

**Comment by xylar** [comment]
Hmm, seeing:
```
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      insertion, deletion, substitution = kwargs.get("weights", (1, 1, 1))
      weights.insert_cost = insertion
      weights.delete_cost = deletion
      weights.replace_cost = substitution
      self.context = weights
      self.dtor = KwargsDeinit
                  ^
  ------------------------------------------------------------

  /home/conda/feedstock_root/build_artifacts/rapidfuzz_1696620205910/work/src/rapidfuzz/distance/metrics_cpp.pyx:307:16: Cannot assign type 'void (RF_Kwargs *) except *' to 'void (*)(RF_Kwargs *) noexcept'. Exception values are incompatible. Suggest adding 'noexcept' to type 'void (RF_Kwargs *) except *'.
```
@maxbachmann, can you have a look and see if you have a suggestion?  No rush.

**Comment by maxbachmann** [comment]
Not sure in which format conda requires the patch, but it still requires the following change:
```
diff --git a/src/rapidfuzz/distance/metrics_cpp.pyx b/src/rapidfuzz/distance/metrics_cpp.pyx
index 3d4940b..0a89219 100644
--- a/src/rapidfuzz/distance/metrics_cpp.pyx
+++ b/src/rapidfuzz/distance/metrics_cpp.pyx
@@ -289,7 +289,7 @@ def levenshtein_opcodes(s1, s2, *, processor=None, score_hint=None):
     ops.editops = levenshtein_editops_func(s1_proc.string, s2_proc.string, c_score_hint)
     return ops.as_opcodes()
 
-cdef void KwargsDeinit(RF_Kwargs* self):
+cdef void KwargsDeinit(RF_Kwargs* self) noexcept:
     free(<void*>self.context)
 
 cdef bool LevenshteinKwargsInit(RF_Kwargs* self, dict kwargs) except False:
```
with this I can build it successfully on my local machine

File: recipe/0001-fix-cmake-cython.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+-    add_cython_target(${_name} CXX)
+-    set(${_name}
+-        ${_name}
+-        PARENT_SCOPE)
+-  endif()
++  find_package(Cython REQUIRED)
++  # should use target_include_directories once this is supported by
++  # scikit-build
++  include_directories(${RF_BASE_DIR}/rapidfuzz
++                      ${RF_BASE_DIR}/rapidfuzz/distance)
```
Comment (maxbachmann) [inline recipe/0001-fix-cmake-cython.patch:24]: here the second include directory is not needed, but it doesn't hurt either.

**Comment by maxbachmann** [comment]
I had another thought about this and it would probably make sense to make a quick 2.15.2 release of the library with the fix, so it works for the poetry version from pypi as well.

**Comment by maxbachmann** [comment]
I started the release procedure. This takes a couple hours to build all the binary wheels. Afterwards we can base this on rapidfuzz 2.15.2 without any patches.

**Comment by xylar** [comment]
I agree, that sounds like a much better way to handle this.  Thank you very much!

**Comment by xylar** [comment]
Closing this in favor of #155

**Pull request closed without merging**